src/hotspot/cpu/x86/stubGenerator_x86_64_poly_mont.cpp

```diff
@@ -0,0 +1,429 @@
+  __ evmovdquq(mask52, allLimbs, ExternalAddress(p256_mask52()), false, Assembler::AVX_512bit, rscratch);
+
+  // M = load(*modulus_p256)
+  __ evmovdquq(modulus, allLimbs, ExternalAddress(modulus_p256()), false, Assembler::AVX_512bit, rscratch);
+
+  // A = load(*aLimbs)
+  __ evmovdquq(A, Address(aLimbs, 8), Assembler::AVX_256bit);                          // Acc1 = load(*a)
+  __ evpermq(A, allLimbs, shift1L, A, false, Assembler::AVX_512bit);
+  __ movq(T, Address(aLimbs, 0));
+  __ evporq(A, A, T, Assembler::AVX_512bit);
+
+  // Acc1 = 0
+  __ vpxorq(Acc1, Acc1, Acc1, Assembler::AVX_512bit);
+  for (int i = 0; i< 5; i++) {
+      // Acc2 = 0
+      __ vpxorq(Acc2, Acc2, Acc2, Assembler::AVX_512bit);
+
+      // B = replicate(bLimbs[i])
+      __ vpbroadcastq(B, Address(bLimbs, i*8), Assembler::AVX_512bit);
+
+      // Acc1 += A * B
+      __ evpmadd52luq(Acc1, A, B, Assembler::AVX_512bit);
+
+      // Acc2 += A *h B
+      __ evpmadd52huq(Acc2, A, B, Assembler::AVX_512bit);
+
+      // N = replicate(Acc1[0])
+      __ vpbroadcastq(N, Acc1, Assembler::AVX_512bit);
+
+      // Acc1 += M *  N
+      __ evpmadd52luq(Acc1, modulus, N, Assembler::AVX_512bit);
+
+      // Acc2 += M *h N
+      __ evpmadd52huq(Acc2, modulus, N, Assembler::AVX_512bit);
+
+      if (i == 4) break;
+
+      // Combine high/low partial sums Acc1 + Acc2
+
+      // carry = Acc1[0] >> 52
+      __ evpsrlq(carry, limb0, Acc1, 52, true, Assembler::AVX_512bit);
+
+      // Acc2[0] += carry
+      __ evpaddq(Acc2, limb0, carry, Acc2, true, Assembler::AVX_512bit);
+
+      // Acc1 = Acc1 shift one q element >>
+      __ evpermq(Acc1, allLimbs, shift1R, Acc1, false, Assembler::AVX_512bit);
+
+      // Acc1 = Acc1 + Acc2
+      __ vpaddq(Acc1, Acc1, Acc2, Assembler::AVX_512bit);
+  }
+
+  // Last Carry round: Combine high/low partial sums Acc1<high_bits> + Acc1 + Acc2
+  // carry = Acc1 >> 52
+  __ evpsrlq(carry, allLimbs, Acc1, 52, true, Assembler::AVX_512bit);
+
+  // Acc1 = Acc1 shift one q element >>
+  __ evpermq(Acc1, allLimbs, shift1R, Acc1, false, Assembler::AVX_512bit);
+
+  // Acc1  = mask52(Acc1)
+  __ evpandq(Acc1, Acc1, mask52, Assembler::AVX_512bit); // Clear top 12 bits
+
+  // Acc2 += carry
+  __ evpaddq(Acc2, allLimbs, carry, Acc2, true, Assembler::AVX_512bit);
+
+  // Acc1 = Acc1 + Acc2
+  __ vpaddq(Acc1, Acc1, Acc2, Assembler::AVX_512bit);
+
+  // output to rLimbs (1 + 4 limbs)
+  __ movq(Address(rLimbs, 0), Acc1);
+  __ evpermq(Acc1, k0, shift1R, Acc1, true, Assembler::AVX_512bit);
+  __ evmovdquq(Address(rLimbs, 8), k0, Acc1, true, Assembler::AVX_256bit);
+}
+
+address StubGenerator::generate_intpoly_montgomeryMult_P256() {
+  __ align(CodeEntryAlignment);
+  StubCodeMark mark(this, "StubRoutines", "intpoly_montgomeryMult_P256");
+  address start = __ pc();
+  __ enter();
+
+  // Save all 'SOE' registers
+  __ push(rbx);
+  #ifdef _WIN64
+  __ push(rsi);
+  __ push(rdi);
+  #endif
+  __ push(r12);
+  __ push(r13);
+  __ push(r14);
+  __ push(r15);
+
```

Comment:
No need to save/restore  rbx, r12, r14, r15.  Only r13 is used as temp in montgomeryMultiply(aLimbs, bLimbs, rLimbs). That too could be easily changed to r8.

Reply:
Seems I forgot to completely cleanup, thanks! (Originally copied from poly1305 stub)